src/js/messaging/containers/Compose.jsx

```diff
@@ -1,64 +1,79 @@
 import React from 'react';
 import { connect } from 'react-redux';
 
-import { messageCategories, composeMessageErrors, composeMessagePlaceholders } from '../config';
+import {
+  composeMessageErrors,
+  composeMessagePlaceholders,
+  messageCategories
+} from '../config';
+
 import MessageCategory from '../components/compose/MessageCategory';
 import MessageFrom from '../components/compose/MessageFrom';
 import MessageSubject from '../components/compose/MessageSubject';
 import MessageRecipient from '../components/compose/MessageRecipient';
+import MessageSend from '../components/compose/MessageSend';
+
 import {
-  setCategory,
-  setRecipient,
-  setSubject,
+  confirmDelete,
+  saveMessage,
+  sendMessage,
+  setMessageField,
   setSubjectRequired,
-  fetchRecipients
+  fetchRecipients,
+  fetchSenderName
 } from '../actions/compose';
 
 class Compose extends React.Component {
   constructor() {
     super();
     this.handleCategoryChange = this.handleCategoryChange.bind(this);
-    this.handleSubjectChange = this.handleSubjectChange.bind(this);
     this.handleRecipientChange = this.handleRecipientChange.bind(this);
+    this.handleSubjectChange = this.handleSubjectChange.bind(this);
   }
 
   componentWillMount() {
+    this.props.fetchSenderName();
     this.props.fetchRecipients();
   }
 
   handleCategoryChange(valueObj) {
-    this.props.setCategory(valueObj);
+    this.props.setMessageField('message.category', valueObj);
     this.props.setSubjectRequired(valueObj);
   }
 
   handleSubjectChange(valueObj) {
-    this.props.setSubject(valueObj);
+    this.props.setMessageField('message.subject.value', valueObj);
   }
 
   handleRecipientChange(valueObj) {
-    this.props.setRecipient(valueObj);
+    this.props.setMessageField('message.recipient', valueObj);
```

Comment:
I think I preferred the old way, even though it creates a larger set of names to keep track of. This seems to be too closely coupled with the usage of the specific `set` function that updates the store, as the meaning of it isn't fully clear unless you follow the stack to that call and know that's how it works.